contracts/cw3-dao/src/contract.rs

```diff
@@ -384,6 +387,25 @@ pub fn execute_update_config(
         .add_attribute("sender", info.sender))
 }
 
+pub fn execute_update_staking_contract(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    new_staking_contract: Addr,
+) -> Result<Response<Empty>, ContractError> {
+    // Only contract can call this method
+    if env.contract.address != info.sender {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    // Replace the existing staking contract
+    STAKING_CONTRACT.save(deps.storage, &new_staking_contract)?;
```

Comment:
Let's validate the address here to prevent a DAO from getting locked up.

```suggestion
    let new_staking_contract = deps.api.addr_validate(new_staking_contract.as_str())?;

    // Replace the existing staking contract
    STAKING_CONTRACT.save(deps.storage, &new_staking_contract)?;
```